apps/src/lib/levelbuilder/lesson-editor/FindProgrammingExpressionDialog.jsx

```diff
@@ -0,0 +1,217 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import Button from '@cdo/apps/templates/Button';
+import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+
+import LessonEditorDialog from './LessonEditorDialog';
+
+const SearchForm = function(props) {
+  return (
+    <form className="form-search">
+      <input
+        type="text"
+        className="input-large search-query"
+        onChange={props.onSearch}
+        placeholder="search by name"
+      />
+      <select
+        className="input-small search-query"
+        style={{marginLeft: 4}}
+        onChange={props.onFilter}
+      >
+        <option value="">filter</option>
+        {/* TODO: these should probably be sourced from somewhere, rather than hardcoded */}
+        <option value="applab">Applab</option>
+        <option value="gamelab">Gamelab</option>
+        <option value="spritelab">Spritelab</option>
+        <option value="weblab">Weblab</option>
+      </select>
+    </form>
+  );
+};
+
+SearchForm.propTypes = {
+  onSearch: PropTypes.func.isRequired,
+  onFilter: PropTypes.func.isRequired
+};
+
+const ProgrammingExpressionTable = function(props) {
+  if (
+    !props.programmingExpressions ||
+    props.programmingExpressions.length === 0
+  ) {
+    return null;
+  }
+
+  // TODO what should we actually do here?
+  const defaultColor = '#0094ca';
+
+  // TODO implement pagination
```

Comment:
[tracked here](https://codedotorg.atlassian.net/browse/PLAT-880)

Reply:
this seems like a great stopping point. if we add scrolling/pagination, we may want to move this onto reactabular.

Reply:
one minor concern here is that we're planning to copy this approach in rapid succession for (1) adding ProgrammingExpressions to lessons and (2) adding standards to lessons, so we may end up duplicating this somewhat temporary UI a bit. I think that is probably still the best path to hitting our deadline, though.

Reply:
My expectation is that we'd reuse this component for those uses, not copy it

Reply:
agreed for adding programming expressions to lessons. I'll try to reuse or extract something shared when adding standards to lessons.

Reply:
I have something that might be pretty easy to set up for pagination that we used for the add level dialog. I can take a look at adding it when I use the component for adding programming expressions to a lesson